pkg/cmd/registry/artifacts/metadata/metadata.go

```diff
@@ -0,0 +1,136 @@
+package metadata
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	flagutil "github.com/redhat-developer/app-services-cli/pkg/cmdutil/flags"
+	"github.com/redhat-developer/app-services-cli/pkg/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/dump"
+	"github.com/redhat-developer/app-services-cli/pkg/iostreams"
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/serviceregistry/registryinstanceerror"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	"github.com/redhat-developer/app-services-cli/internal/config"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/factory"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/artifacts/util"
+
+	"github.com/redhat-developer/app-services-cli/pkg/logging"
+)
+
+type Options struct {
+	artifact     string
+	group        string
+	outputFormat string
+
+	registryID string
+
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Logger     func() (logging.Logger, error)
+	Connection factory.ConnectionFunc
+	localizer  localize.Localizer
+}
+
+func NewMetadataCommand(f *factory.Factory) *cobra.Command {
+	opts := &Options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Logger:     f.Logger,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "metadata",
+		Short: "Get artifact metadata",
+		Long: `
+Gets the metadata for an artifact in the service registry. 
+The returned metadata includes both generated (read-only) and editable metadata (such as name and description).
+`,
+		Example: `
+## Get latest artifact metadata for default group
+rhoas service-registry artifacts metadata my-artifact
+
+## Get latest artifact metadata for my-group group
+rhoas service-registry artifacts metadata my-artifact --group mygroup 
+		`,
+		Args: cobra.RangeArgs(0, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.artifact = args[0]
+			}
+
+			if opts.registryID != "" {
+				return runGet(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return fmt.Errorf("No service Registry selected. Use 'rhoas service-registry use' to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+			return runGet(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.artifact, "artifact", "a", "", "Id of the artifact")
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Group of the artifact")
+	cmd.Flags().StringVarP(&opts.registryID, "registryId", "", "", "Id of the registry to be used. By default uses currently selected registry")
```

Comment:
Remove camel casing for `registryID` flag

Reply:
How it should look like?

Reply:
Yes, kebab-case.

Reply:
Should we call it `instance-id`? We could use this across all instance types..such as Kafka, Connector etc.

Reply:
Good idea. Changing